src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +251,230 @@ def inner(tree, path, root_path):
+    wildcard: Optional[bool] = False,
+    match_whole_word: Optional[bool] = False,
+    match_case: Optional[bool] = True,
+):
+    """Search for a word through the Fluent's object hierarchy.
+
+    Parameters
+    ----------
+    search_string: str
+        The word to search for. Semantic search is default.
+    language: str
+        The language for the semantic search.
+        English is default for the semantic search.
+        ISO 639-3 code of the language to be used for semantic search.
+        See `https://omwn.org/omw1.html` for the list of supported languages.
+        The default value is `eng` for English language.
+    wildcard: bool
+        Whether to use wildcard pattern. If ``True`` will match wildcard pattern based on ``fnmatch`` module and
+        will turn off semantic matching.
+    match_whole_word: bool
+        Whether to get exact match. If ``True`` will match exact string and will turn off semantic matching.
+    match_case: bool
+        Whether to match case. If ``True`` will match case-insensitive case.
+
+    Raises
+    ------
+    ValueConflict
+        If both ``wildcard`` and ``match_whole_word`` are ``True``.
+
+    Examples
+    --------
+    >>> import ansys.fluent.core as pyfluent
+    >>> pyfluent.search("font", match_whole_word=True)
+    >>> pyfluent.search("Font")
+    >>> pyfluent.search("iter*", wildcard=True)
+    >>> pyfluent.search("读", language="cmn")   # search 'read' in Chinese
+    The most similar API objects are:
+    <solver_session>.file.read (Command)
+    <solver_session>.file.import_.read (Command)
+    <solver_session>.mesh.surface_mesh.read (Command)
+    <solver_session>.tui.display.display_states.read (Command)
+    <meshing_session>.tui.display.display_states.read (Command)
+    """
+    if wildcard and match_whole_word:
+        raise ValueConflict()
+    elif language and match_whole_word:
+        warnings.warn(
+            "``match_whole_word=True`` will match exact string and will turn off semantic matching.",
+            UserWarning,
+        )
+    elif match_whole_word:
+        warnings.warn(
+            "``match_whole_word=True`` will turn off semantic and wildcard matching.",
+            UserWarning,
+        )
+
+    api_object_names = get_api_tree_file_name(name=True)
+    names_txt = open(api_object_names, "r")
+    names = [name.replace("\n", "") for name in names_txt.readlines()]
+    synset_1 = wn.synsets(search_string, lang=language)
+
+    if wildcard and not match_whole_word:
+        queries = _process_wildcards(search_string, names)
+    elif synset_1 and not match_whole_word and not wildcard:
+        similar_keys = set()
+        for name in names:
+            synset_2 = wn.synsets(name, lang="eng")
+            for s1 in synset_1:
+                for s2 in synset_2:
+                    name_s1 = s1.name().split(".")[0]
+                    name_s2 = s2.name().split(".")[0]
+                    if search_string in name_s2 or name_s1 in name_s2:
```

Comment:
I need comments. Badly.

Reply:
Example - 

```
`s1.name() = "font.n.01"

Where, 
font - required match
n - noun (part of speech tag)
01 - word type tag
```

We are extracting `font` from `s1.name()` and similar is for `s2.name()`

Reply:
Your response shows that s1 and s2 have meanings that should be reflected in the variable names. That's how the code starts to comment itself. Over and above that, I suggest that additional comments (based on the above description) might still be helpful, but clear naming alone goes a long way to making comments redundant.

Reply:
Yes, I agree. We have updated the variable names. Thank you.